src/euphorie/client/profile.py

```diff
@@ -341,33 +370,50 @@ def ProfileQuestions(self):
         - ``id``: object id of the question
         - ``title``: title of the question
         - ``question``: question about the general occurance
-        - ``label_multiple_present``: question about single or multiple occurance
+        - ``label_multiple_present``: question about single or multiple
+          occurance
         - ``label_single_occurance``: label for single occurance
         - ``label_multiple_occurances``: label for multiple occurance
         """
         return _questions(self.context)
 
+    @property
+    def session(self):
+        return api.content.get_view(
+            'webhelpers',
+            self.context,
+            self.request,
+        ).session
+
     def update(self):
         lang = getattr(self.request, 'LANGUAGE', 'en')
         if "-" in lang:
             elems = lang.split("-")
             lang = "{0}_{1}".format(elems[0], elems[1].upper())
-        self.message_required = translate(_(
-            u"message_field_required", default=u"Please fill out this field."),
-            target_language=lang)
+        self.message_required = translate(
+            _(
+                u"message_field_required",
+                default=u"Please fill out this field."
+            ),
+            target_language=lang
+        )
         survey = aq_inner(self.context)
         self.profile_questions = self.ProfileQuestions()
-        self.session = SessionManager.session
         self.current_profile = extractProfile(survey, SessionManager.session)
```

Comment:
I'm not sure any more, but why is `SessionManager.session` used here instead of `self.session`?

Reply:
I think it is something like a default... But now session is property, that means that it is evaluated whenever you need it and you will be more confident of its origin.